src/package_manager/publish.ts

```diff
@@ -55,12 +55,24 @@ export default async function publish(proj: Project) {
   if (untracked.code === 0) {
     const untrackedList = untracked.all
       .split("\n")
-      .filter((line) => line.trim())
-      .map((line) => "/" + line.trim()); // add "/" to the begining to specify files/dirs relative to .npmignore
+      .filter((line) => {
+        if (!line.trim()) return false;
+        // if the project uses default icon, just ignore all uploaded
+        // icons under .webinizer/icons folder, otherwise ignore other
+        // uploaded icons and just kept the one used by the project.
+        if (proj.config.img?.isUploaded) {
```

Comment:
Also need to consider demo project case - default icon image is uploaded and tracked in git (not in the untracked file list).